blog/content/post/2022-12-22-fobnail_application.md

```diff
@@ -0,0 +1,127 @@
+---
+title: Fobnail Token - example use case
+abstract: 'TBD'
+cover: /covers/usb_token.png
+author: krystian.hebel
+layout: post
+published: true
+date: 2022-12-22
+archives: "2022"
+
+tags:
+  - fobnail
+  - tpm
+  - attestation
+  - linux
+categories:
+  - Security
+
+---
+
+# About the Fobnail project
+
+Fobnail is a project that aims to provide a reference architecture for building
+offline integrity measurement verifiers on the USB device (Fobnail Token) and
+attesters running in Dynamically Launched Measured Environments (DLME). It
+allows the Fobnail owner to verify the trustworthiness of the running system
+before performing any sensitive operation. This project was founded by [NlNet
+Foundation](https://nlnet.nl/). More information about the project can be found
+in the [Fobnail documentation](https://fobnail.3mdeb.com/). Also, make sure to
+read other posts related to this project by visiting
+[fobnail](https://blog.3mdeb.com/tags/fobnail/) tag.
+
+# Scope of current phase
+
+This phase focused on using Fobnail in a real-life use case, namely using it to
+access LUKS2 decryption key if and only if the PCR measurements are valid. In
+this post only high-level usage will be described, if you want to know what
+happens under the hood or want to modify it to your needs see
+[documentation](https://fobnail.3mdeb.com/examples/disk_encryption/).
+
+# Changes
+
+In this phase we completely switched the architecture around. Now the CoAP
+server is located on Fobnail Token, instead of each of PC applications. This
+way we could implement [API endpoints](https://fobnail.3mdeb.com/fobnail-api/),
+including [Fobnail Token Services (FTS)](https://fobnail.3mdeb.com/fobnail-api/#fobnail-token-services),
+which allows for more elaborate applications without having to re-flash Token
+firmware each time.
+
+As of now, only storage services are fully defined in FTS. Few other endpoints
+for working with cryptographic keys were listed, but they are yet to be
+implemented. There is also an endpoint `/api/version` which reports all versions
+of API supported by Token - the version will be increased only when
+backward-incompatible changes are done. **TBD**
+
+There are now two separate Attester applications - one with provisioning code,
+and one without. Former should be used just for initial provisioning, and the
+latter for normal use. We hope that by simplifying more frequently used form we
+can reduce the surface of attack. It also helps with rare cases when users
```

Comment:
I don't understand that sentence. What "form" we referring to?

Reply:
I've changed the wording.